### PR TITLE
#199 prevent application change after adding ws resource

### PR DESCRIPTION
**Pull request by Muhaddatha**
#199 
Prevent open application from changing when a new resource is added to the workspace.

### CODE REVIEW

**Comment by Muhaddatha** [comment]
The bug happens you open a workspace with a specific app that is different from the app of the last opened workspace resource:
1. When adding a new resource to the workspace.
2. When removing a resource from a workspace.

In both scenarios adding or removing a resource triggers the refreshing on the workspace. This refreshing causes the workspaceFrame to re-render. During its re-render, it uses the last opened resource's app to decide what application to open on the right panel. We wanted the original app to not change even if we changed the workspace by adding to or removing from it. The only time that we want the application to change is when we click on a workspace resource.

So, I thought a solution would be to figure out what caused the workspace refresh. If the workspace refresh was caused by a change in the workspace resources array length and not by clicking on a resource, I prevented the workspace frame from loading a new application.

Two different ways I used to reproduce the bug:
1. Open a workspace with a selected app that is different to the opened resource's type. For example, if the last opened resource's type is nwb, open the workspace with netpyne or jupyter. Next, add a resource to the workspace from an OSB repository. This would cause the workspace frame to open the nwbexplorer application.
2. Repeat the above steps but instead of adding a resource, remove one from the list.

**Review by filippomc — CHANGES_REQUESTED** [review]
The number of resources may not be a reliable indicator of any intent. Using the click event and useEffect to trigger the resource loading in place of the render could be a more stable solution

File: applications/osb-portal/src/components/workspace/drawer/WorkspaceResourceBrowser.tsx [hunk 1]
```diff
@@ -27,6 +28,9 @@ const openFileResource = (resource: WorkspaceResource, refreshWorkspace: any) =>
   return workspaceResourceService.workspacesControllerWorkspaceResourceOpen(resource.id).then(() => {
     const iFrame: HTMLIFrameElement = document.getElementById("workspace-frame") as HTMLIFrameElement;
     iFrame.contentWindow.postMessage(fileName, '*');
+    console.log('resource', resource);
+    const domain = getBaseDomain();
+    iFrame.src = `//${resource.type.application.subdomain}.${domain}/hub/spawn/hub/spawn//${resource.workspaceId}${resource.type}`;
```
Comment (filippomc) [inline applications/osb-portal/src/components/workspace/drawer/WorkspaceResourceBrowser.tsx:33]: why is hub/spawn doubled?

Comment (Muhaddatha) [reply applications/osb-portal/src/components/workspace/drawer/WorkspaceResourceBrowser.tsx:33]: It was a typo, I'll remove the extra one.

**Pull request merged**